### ISSUE TITLE
Last upgrade breaks some documents []

### ISSUE THREAD

**Issue by angelog0** [latex3/hyperref#170]
Yesterday I did an update of TeXLive (2020) and I was able to rebuild my documents. This morning (2021.02.06) I did another update and this STC does not build any more:

```
$ cat stc-bug-20210206.tex
\documentclass{article}

\usepackage{amsfonts}
\usepackage{hyperref}

\newcommand{\numberset}{\mathbb}
\newcommand{\C}{\numberset{C}}

\begin{document}
See this $\C$.
\end{document}

## Build with:

$ pdflatex stc-bug-20210206.tex
This is pdfTeX, Version 3.14159265-2.6-1.40.21 (TeX Live 2020) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(./stc-bug-20210206.tex
LaTeX2e <2020-10-01> patch level 4
L3 programming layer <2021-02-02>
(/opt/texlive/texmf-dist/tex/latex/base/article.cls
Document Class: article 2020/04/10 v1.4m Standard LaTeX document class
(/opt/texlive/texmf-dist/tex/latex/base/size10.clo))
(/opt/texlive/texmf-dist/tex/latex/amsfonts/amsfonts.sty)
(/opt/texlive/texmf-dist/tex/latex/hyperref/hyperref.sty
(/opt/texlive/texmf-dist/tex/generic/ltxcmds/ltxcmds.sty)
(/opt/texlive/texmf-dist/tex/generic/iftex/iftex.sty)
(/opt/texlive/texmf-dist/tex/generic/pdftexcmds/pdftexcmds.sty
(/opt/texlive/texmf-dist/tex/generic/infwarerr/infwarerr.sty))
(/opt/texlive/texmf-dist/tex/latex/graphics/keyval.sty)
(/opt/texlive/texmf-dist/tex/generic/kvsetkeys/kvsetkeys.sty)
(/opt/texlive/texmf-dist/tex/generic/kvdefinekeys/kvdefinekeys.sty)
(/opt/texlive/texmf-dist/tex/generic/pdfescape/pdfescape.sty)
(/opt/texlive/texmf-dist/tex/latex/hycolor/hycolor.sty)
(/opt/texlive/texmf-dist/tex/latex/letltxmacro/letltxmacro.sty)
(/opt/texlive/texmf-dist/tex/latex/auxhook/auxhook.sty)
(/opt/texlive/texmf-dist/tex/latex/kvoptions/kvoptions.sty)
(/opt/texlive/texmf-dist/tex/latex/hyperref/pd1enc.def)
(/opt/texlive/texmf-dist/tex/latex/hyperref/hyperref-langpatches.def)
(/opt/texlive/texmf-dist/tex/generic/intcalc/intcalc.sty)
(/opt/texlive/texmf-dist/tex/generic/etexcmds/etexcmds.sty)
(/opt/texlive/texmf-dist/tex/latex/hyperref/puenc.def)
(/opt/texlive/texmf-dist/tex/latex/url/url.sty)
(/opt/texlive/texmf-dist/tex/generic/bitset/bitset.sty
(/opt/texlive/texmf-dist/tex/generic/bigintcalc/bigintcalc.sty))
(/opt/texlive/texmf-dist/tex/latex/base/atbegshi-ltx.sty))
(/opt/texlive/texmf-dist/tex/latex/hyperref/hpdftex.def
(/opt/texlive/texmf-dist/tex/latex/base/atveryend-ltx.sty)
(/opt/texlive/texmf-dist/tex/latex/rerunfilecheck/rerunfilecheck.sty
(/opt/texlive/texmf-dist/tex/generic/uniquecounter/uniquecounter.sty)))

! LaTeX Error: Command \C already defined.
               Or name \end... illegal, see p.192 of the manual.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...

l.7 \newcommand{\C}{\numberset{C}}

?
```

Commenting out  the usage of `hyperref` package, works as expected. This morning there was the update of `hyperref`...

A work around is to change the `newcommand` name. 

Shouold really I do that? or is there some other issue?

Thanks.

**Comment by u-fischer** [comment]
This a side effect of setting the default to pdflatex to unicode. You would have gotten the same error in older versions if you would have used `\usepackage[unicode]{hyperref}`, or with the unicode engines lualatex and xelatex. 

But we will discuss if one can remove this command, but it is also used in cyrillic, so it can't be done directly. 

As a work-around you can simply use \renewcommand for now.  Generally I would avoid 1-letter commands for document commands, there are often used for accents and so clashes are frequent.

**Comment by angelog0** [comment]
OK, I will adopt `\newcommand{\CSet}{\numberset{C}}`...

Thanks

**Comment by cheuberg** [comment]
This change breaks a lot of my TeX files; I think that the use of `\C` for the complex numbers is very widespread (and useful). 

> Generally I would avoid 1-letter commands for document commands, there are often used for accents and so clashes are frequent.

IMHO 1-letter-commands which have not been accents for many years should be available, a change now in 2021 after so many years should at least be optional.

**Comment by u-fischer** [comment]
@cheuberg  We just changed hyperref  and will upload today. It will define the cyrillic (including \C) commands only if \CYRDZE is defined. So documents not using cyrillic should work fine again.

**Comment by cheuberg** [comment]
@u-fischer Thank you.

**Comment by u-fischer** [comment]
hyperref 7.00h  has been uploaded.

**Comment by fstengel** [comment]
Sorry to reopen this: there are still some issues with one letter commands. As @cheuberg said: everything worked well until 7.00g and has been so for ages. The issue is that one finds it rather hard to know which commands are preempted by hyperref and why PU encoding is loaded by default when it was not until 7.00h. Using unicode=false does nothing in that respect: puenc.def is always loaded... 

In my documents, the offending command is \G. Not that I mind too much : I can overwrite it with a \def in a style sheet, but still annoying.

**Comment by fstengel** [comment]
@u-fischer Thanks a lot. Your kind answer helped me understand what is happening.

The idea with moving the handling of the non OT1/T1 encoding accent commands to a separate file is interesting.  One could even imagine that hyperref catches the basic encoding (by looking up \encodingdefault?) and automagically loads these files as needed. As far as I can understand, that is already done for the cyrillic encodings by checking the existence of \CYRDZE

**Comment by u-fischer** [comment]
no sorry. The idea is to make thinks simpler, not to add a new layer of conditionals on it. Beside this it would be rather confusing for users if a document suddenly errors if a language (and its encoding) is added, so such dependencies should be kept at a minimum.
